src/main/java/minijava/lexer/TaggedWithLocation.java

```diff
@@ -0,0 +1,42 @@
+package minijava.lexer;
+
+import java.util.Iterator;
+
+public class TaggedWithLocation<T> {
+
+  public final Location loc;
+  public final T tagged;
+
+  public TaggedWithLocation(Location loc, T tagged) {
+    this.loc = loc;
+    this.tagged = tagged;
+  }
+
+  public static Iterator<TaggedWithLocation<Byte>> tagBytes(Iterator<Byte> bytes) {
+    return new Iterator<TaggedWithLocation<Byte>>() {
+
+      private int currentLine = 1;
+      private int currentColumn = 0;
+
+      @Override
+      public boolean hasNext() {
+        return bytes.hasNext();
+      }
+
+      @Override
+      public TaggedWithLocation<Byte> next() {
+        Location l = new Location(currentLine, currentColumn);
+        byte b = bytes.next();
+
+        if (b == '\n') {
+          currentLine++;
+          currentColumn = 0;
+        } else {
+          currentColumn++;
+        }
+
+        return new TaggedWithLocation<>(l, b);
```

Comment:
Yep... Just throwing stuff at the wall and see what sticks... No need to review at this point.